### PR TITLE
Add rule: Close ticket window on close ticket

### LINKED ISSUE #9: New Rule: Close window on ticket closed
Proposed spec: Automatically close the ticket window (close the tab) once the ticket status has been updated or edited to "Closed."

This is to avoid having to close the window after you are done with the ticket.

> User requested

### PR DESCRIPTION
**Pull request by ekcom**
Resolves #9

### CODE REVIEW

**Comment by ekcom** [comment]
This may be `wontfix`. Closing windows without being the one opening them looks impossible.

[MDN window.close](https://developer.mozilla.org/en-US/docs/Web/API/Window/close):
> This method can only be called on windows that were opened by a script using the window.open() method, or on top-level windows that have a single history entry.

**Comment by ekcom** [comment]
Aha, a workaround!

[Background scripts can close tabs](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/remove)

File: src/utils/services/middleman.ts [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+            return {
+                status: "error",
+                message: "bad",
+            } satisfies CompletedRequest;
+        } else {
+            // ok to handle. valid at this point.
+            // request.message.data may be undefined; this is ok
+            return await handle(sender.tab, request.message.request, request.message.data);
+        }
+    } else if (stringBeginsWith(sender.url, BASE_URL)) { // todo check destination as well (maybe sniff packets too)
```
Comment (edthedev) [inline src/utils/services/middleman.ts:27]:
```suggestion
    } else if (stringBeginsWith(sender.url, BASE_URL)) && (request.to === PIPELINE_TICKET_TO_CEREBRO ) {
```

File: src/utils/services/middleman.ts [hunk 2]
```diff
@@ -1,16 +1,30 @@
-import browser from "webextension-polyfill";
+import * as browser from "webextension-polyfill";
 import { findBestTab } from "../findBestTab";
 import { log } from "utils/logger";
 import { stringBeginsWith } from "utils/stringParser";
 import { BASE_URL, CEREBRO_URL } from "config";
+import { PIPELINE_TO_BACKGROUND_SCRIPT } from "utils/webpage/link/interface";
```
Comment (edthedev) [inline src/utils/services/middleman.ts:6]:
```suggestion
import { PIPELINE_TO_BACKGROUND_SCRIPT, PIPELINE_TICKET_TO_CEREBRO  } from "utils/webpage/link/interface";
```

**Review by edthedev — APPROVED** [review]

File: src/utils/services/middleman.ts (hunk 1 above)
Comment (edthedev) [reply src/utils/services/middleman.ts:27]: Deferring this for later, when there's time to re-test.

File: src/utils/services/middleman.ts (hunk 2 above)
Comment (edthedev) [reply src/utils/services/middleman.ts:6]: Deferring this for later, when there's time to re-test.

**Review by tzturner — APPROVED** [review]

**Pull request merged**